imap_processing/hi/l1a/housekeeping.py

```diff
@@ -0,0 +1,92 @@
+"""Unpack IMAP-Hi housekeeping data."""
+import collections
+import dataclasses
+
+import xarray as xr
+
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.cdf.utils import calc_start_time
+from imap_processing.hi.hi_cdf_attrs import (
+    hi_hk_l1a_attrs,
+    hi_hk_l1a_metadata_attrs,
+)
+
+
+def create_dataset(packets):
+    """Create dataset for each metadata field.
+
+    Parameters
+    ----------
+    packets : list
+        packet list
+
+    Returns
+    -------
+    xr.dataset
+        dataset with all metadata field data in xr.DataArray
+    """
+    metadata_arrays = collections.defaultdict(list)
+    description_arrays = collections.defaultdict(list)
+
+    for data_packet in packets:
+        # Add metadata to array
+        for key, value in data_packet.header.items():
```

Comment:
You can avoid the two for loops by combining the two dictionaries together I think.
```suggestion
        for key, value in (data_packet.header | data_packet.data).items():
```

Reply:
Thank you! Great point.